### PR TITLE
test(scorers): increase conversational.py test coverage from 78% to 81%

### PR DESCRIPTION
**Pull request by shehaaz**
- Add comprehensive tests for async helper function _run_async_in_sync_context
- Add edge case tests for all parsing methods (_parse_knowledge_items, _parse_violations, etc.)
- Add input validation tests for invalid types and empty inputs
- Add tests for all reasoning generation methods across score ranges
- Add tests for data model creation (ConversationTurn, Conversation, KnowledgeItem)
- Add tests for scorer configuration and combined metrics
- Improve error handling coverage for all scorer classes

Total: 22 new test cases added, bringing coverage from 78% to 81% All tests pass (52/52) and code follows project quality standards.

🤖 Generated with [Claude Code](https://claude.ai/code)

# Pull Request

## Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Code refactoring
- [ ] Test improvements

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

- [ ] Unit tests
- [ ] Integration tests
- [ ] Manual testing
- [ ] Performance testing

## Test Configuration

* Python version:
* Operating System:
* Dependencies:

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## Additional Notes

Add any other context about the pull request here.

## Summary by CodeRabbit

* **Tests**
  * Added extensive new unit tests to improve coverage of conversational scoring features, including input validation, edge cases, and reasoning generation for various scoring scenarios. No changes were made to existing test logic.

### CODE REVIEW

File: tests/unit/test_scorers_conversational.py [hunk 1]
```diff
@@ -481,6 +481,439 @@ def test_validate_inputs(self):
...
+        assert "Moderate knowledge retention" in reasoning
+
+        # Test poor score
+        reasoning = scorer._generate_reasoning(0.4, "output", None)
+        assert "Poor knowledge retention" in reasoning
+
+        # Test very poor score
+        reasoning = scorer._generate_reasoning(0.1, "output", None)
+        assert "Very poor knowledge retention" in reasoning
+
```
[the remark this answers is not in the document]
Comment (imshashank) [reply tests/unit/test_scorers_conversational.py:653]: @shehaaz Please look at these comments.

File: tests/unit/test_scorers_conversational.py [hunk 2]
```diff
@@ -481,6 +481,439 @@ def test_validate_inputs(self):
...
+        assert "Moderate relevancy" in reasoning
+
+        # Test poor relevancy
+        reasoning = scorer._generate_relevancy_reasoning(0.4, "output", None)
+        assert "Poor relevancy" in reasoning
+
+        # Test very poor relevancy
+        reasoning = scorer._generate_relevancy_reasoning(0.1, "output", None)
+        assert "Very poor relevancy" in reasoning
+
```
[the remark this answers is not in the document]
Comment (imshashank) [reply tests/unit/test_scorers_conversational.py:714]: @shehaaz Please look at these comments.
